source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -99,6 +98,7 @@ class AuthenticatorImpl : public Logger::Loggable<Logger::Id::jwt>,
   const bool is_allow_failed_;
   const bool is_allow_missing_;
   TimeSource& time_source_;
+  ::google::jwt_verify::Jwt* cache_jwt_;
```

Comment:
My suggestion is:
1) change jwt_ to be the raw pointer as `::google::jwt_verify::Jwt*`
2) rename std::unique_ptr<::google::jwt_verify::Jwt> owned_jwt_;  only used for non cached jwt
3) create a new bool use_cache_jwt_{};



```
Class Authenticator {
...

   std::unique_ptr<::google::jwt_verify::Jwt> owned_jwt_; 
   ::google::jwt_verify::Jwt* jwt_{};
   bool use_cache_jwt_{};
}

If (jwt_cache_is_valid) {
   // use the cached jwt directly, not make a copy
   jwt_ = jwt_cache->Lookup(token);
   if (jwt_) {
     use_cache_jwt_ = true;
     // at Destructor of authenticator need to release it by call
     // jwt_cache->ReleaseLookup(token);
   }
} 

if (!jwt_) {
  owned_jwt_ = std::make_unique<::google::jwt_verify::Jwt>();
 status = owned_jwt_->parseFromString(curr_token_->token());
  jwt_ = owned_jwt.get();
}
```

Reply:
Calling `release` on dtor of the authenticator will only release the last used token. so, calling after `find` if the cache found.